src/main/java/org/broadinstitute/hellbender/utils/codecs/sampileup/SAMPileupCodec.java

```diff
@@ -0,0 +1,343 @@
+package org.broadinstitute.hellbender.utils.codecs.sampileup;
+
+import htsjdk.tribble.AsciiFeatureCodec;
+import htsjdk.tribble.exception.CodecLineParsingException;
+import htsjdk.tribble.readers.LineIterator;
+import htsjdk.tribble.util.ParsingUtils;
+
+import java.util.ArrayList;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static org.broadinstitute.hellbender.utils.codecs.sampileup.SAMPileupFeature.VariantType;
+
+/**
+ * Decoder for SAM pileup data.
+ *
+ * <p>
+ *     From the <a href="http://samtools.sourceforge.net/">SAMTools project documentation</a>:
+ * </p>
+ * <p>The Pileup format was first used by Tony Cox and Zemin Ning at
+ *     the Sanger Institute. It describes the base-pair information at each chromosomal position. This format
+ *     facilitates SNP/indel calling and brief alignment viewing by eye. Note that the pileup program has been replaced
+ *     in Samtools by mpileup, which produces a slightly different output format by default.
```

Comment:
Do we handle single sample mpileups?  We definitely should since it's the standard now.